Example/Frameworks/Database/Logger.swift

```diff
@@ -27,35 +27,133 @@ import UIKit
 import Willow
 
 extension LogLevel {
-    /// Custom log level for SQL log messages with a bitmask of `1 << 8`.
-    public static let sql = LogLevel(rawValue: 0b00000000_00000000_00000001_00000000)
+    // Namespace our custom levels
+    public enum Database {
+        /// Custom log level for SQL log messages with a bitmask of `1 << 8`.
+        public static var sql = LogLevel(rawValue: 0b00000000_00000000_00000001_00000000)
+    }
 }
 
 // MARK: -
 
+// Extend Logger to have `sql` log level functions.
 extension Logger {
-    func sql(_ message: @autoclosure @escaping () -> String) {
-        logMessage(message, with: .sql)
+    func sql(_ message: @autoclosure @escaping () -> Message) {
+        logMessage(message, with: LogLevel.Database.sql)
     }
 
-    func sql(_ message: @escaping () -> String) {
-        logMessage(message, with: .sql)
+    func sql(_ message: @escaping () -> Message) {
+        logMessage(message, with: LogLevel.Database.sql)
+    }
+}
+
+// Extend Logger optional support to have `sql` log level functions.
+extension Optional where Wrapped == Logger {
+    func sql(_ message: @autoclosure @escaping () -> Message) {
+        guard case let .some(log) = self else { return }
+        log.logMessage(message, with: LogLevel.Database.sql)
+    }
+
+    func sql(_ message: @escaping () -> Message) {
+        guard case let .some(log) = self else { return }
+        log.logMessage(message, with: LogLevel.Database.sql)
     }
 }
```

Comment:
Yes, that is unfortunate. I'd guess custom log levels are not used a whole lot and this won't be a pain point for most users. If that is not true, it might be worth evaluating alternatives.